src/bin/pg_autoctl/monitor.c

```diff
@@ -346,6 +358,217 @@ monitor_node_active(Monitor *monitor,
 }
 
 
+/*
+ * monitor_set_node_replication_settings updates the monitor on the changes
+ * in the node replication settings. Caller should first retrieve
+ * existing values if setting only one parameter.
+ */
+bool
+monitor_set_node_replication_settings(Monitor *monitor, int nodeid,
+									  int candidate_priority, bool replicationQuorum)
+{
+	PGSQL *pgsql = &monitor->pgsql;
+	const char *sql = "SELECT * FROM pgautofailover.set_node_replication_settings($1, $2, $3)";
+	int paramCount = 3;
+	Oid paramTypes[3] = { INT4OID, INT4OID, BOOLOID };
+	const char *paramValues[3];
+	char *candidatePriorityText = intToString(candidate_priority).strValue;
+	char *replicationQuorumText = replicationQuorum ? "true" : "false";
+
+	paramValues[0] = intToString(nodeid).strValue;
+	paramValues[1] = candidatePriorityText;
+	paramValues[2] = replicationQuorumText;
+
+	if (!pgsql_execute_with_params(pgsql, sql,
+								   paramCount, paramTypes, paramValues,
+								   NULL, NULL))
+	{
+		log_error("Failed to update node replication state on node %d"
+				  " for candidate_priority : %s"
+				  " and replication_quorum  : %s",
+				  nodeid, candidatePriorityText, replicationQuorumText);
+
+		/* disconnect from monitor */
+		pgsql_finish(&monitor->pgsql);
+
+		return false;
+	}
+
+	/* disconnect from monitor */
+	pgsql_finish(&monitor->pgsql);
+
+	return true;
+}
+
+/*
+ * monitor_get_node_replication_settings retrieves replication settings
+ * from the monitor.
+ */
+bool
+monitor_get_node_replication_settings(Monitor *monitor, int nodeid,
+									  int *candidate_priority, bool *replicationQuorum)
```

Comment:
We only have two parameters at the moment, though we might add more in the future. I think we should make a `typedef struct` for those parameters and use the struct here rather than every single parameter separately.